Lib/test/test_capi/test_misc.py

```diff
@@ -665,19 +667,46 @@ def test_c_subclass_of_heap_ctype_with_del_modifying_dunder_class_only_decrefs_o
         del subclass_instance
 
         # Test that setting __class__ modified the reference counts of the types
+        #
```

Comment:
A bit of an aside, but I'm skeptical of the value of this test considering the increasing complexity. If we want to make sure we don't decref too many times, we should test that we don't crash. Depending on the exact refcounts feels too likely to break with implementation changes.

Reply:
Agreed. Checking that we don't crash sounds like a better approach to me. Filed https://github.com/python/cpython/issues/127881 to track that.